superset/db_engine_specs/drill.py

```diff
@@ -84,7 +87,7 @@ def modify_url_for_impersonation(
         if impersonate_user and username is not None:
             if url.drivername == "drill+odbc":
                 url.query["DelegationUID"] = username
-            elif url.drivername == "drill+jdbc":
+            elif url.drivername in ["drill+sadrill", "drill+jdbc"]:
                 url.query["impersonation_target"] = username
             else:
-                url.username = username
+                logger.warning("impersonation is not supported for %s", url.drivername)
```

Comment:
IMO there's risk here that the user will *think* impersonation is enabled when using other than `odbc`, `sadrill` or `jdbc`. Would it make sense to rather raise an exception in the `else` if we're trying to impersonate and it's not supported by the driver?

Reply:
Yes, good call.